src/libra_py/dynamics/qtag/QTAG_main.py

```diff
@@ -0,0 +1,229 @@
+"""
+..module:: QTAG_main
+  :platform: Unix, Windows
+  :synopsis: This module is the main body of the QTAG method, responsible for running cycles of the BOT algorithm.
+..moduleauthors :: Matthew Dutra
+"""
+
+import sys
+import os
+from liblibra_core import *
+import util.libutil as comn
+import time
+import datetime
+
+from libra_py import data_outs
+import matplotlib.pyplot as plt
+import numpy as np
+
+from QTAG_config import *
+
+import QTAG_init
+import QTAG_basis
+import QTAG_ham
+import QTAG_mom
+import QTAG_pots
+import nafh_pot
+import QTAG_prop
+import QTAG_calc
+
+#Create output file objects...
+f_traj1=open('gbc1.txt', 'w')
+f_traj2=open('gbc2.txt', 'w')
+f_mom1=open('p1.txt', 'w')
+f_mom2=open('p2.txt', 'w')
+f_a1=open('a1.txt', 'w')
+f_a2=open('a2.txt', 'w')
+f_obs=open('obs.txt', 'w')
+f_wf1=open('wft1.txt', 'w')
+f_wf2=open('wft2.txt', 'w')
+
+#Libra + QTAG checks for input dictionaries...
+comn.check_input(univ,{"dt":0.1,"niter":1,"n_data_out":1,"n_snapshots":1},["ndof","ntraj","mass"])
```

Comment:
i'd try to keep the number of "critical parameters" to the minimum - very often reasonable default numbers could be guessed - e.g. 1 dof, 1 trajectory, 1 (or 2000 units of mass), etc. - just imagine some hypothetical system one would be modeling if they won't provide any input